src/components/tools/geometry-tool/geometry-tool.tsx

```diff
@@ -270,36 +272,38 @@ class GeometryToolComponentImpl extends BaseComponent<IProps, IState> {
           onDragOver={this.handleDragOver}
           onDragLeave={this.handleDragLeave}
           onDrop={this.handleDrop} />,
-      this.renderMenuButton(),
       this.renderRotateHandle()
     ]);
   }
 
-  private renderMenuButton() {
-    const { model, readOnly } = this.props;
+  private renderToolbar() {
     const { board } = this.state;
-    const isSelectedTile = this.stores.ui.isSelectedTile(model);
-    if (!isSelectedTile || readOnly || !board) return;
+    if (!board) return;
+    const { readOnly } = this.props;
+    const content = this.getContent();
 
-    const angleIcon = renderUnicodeCharAsIconElement("∡");
     const selectedObjects = this.getContent().selectedObjects(board);
     const selectedPoints = selectedObjects && selectedObjects.filter(isPoint);
     const selectedPoint = selectedPoints && (selectedPoints.length === 1)
                             ? selectedPoints[0] as JXG.Point : undefined;
     const supportsVertexAngle = selectedPoint && canSupportVertexAngle(selectedPoint);
-    const hasVertexAngle = selectedPoint && !!getVertexAngle(selectedPoint);
-    const vertexAngleText = !hasVertexAngle ? "Show Angle Label" : "Hide Angle Label";
-    const enableLabelAngle = !readOnly && supportsVertexAngle;
+    const hasVertexAngle: boolean = !!selectedPoint && !!getVertexAngle(selectedPoint);
+    const enableVertexAngle: boolean = !readOnly && !!supportsVertexAngle;
+    const enableDelete: boolean = !readOnly && board && content.hasSelection();
+    const enableDuplicate: boolean = !readOnly && board && !!this.getOneSelectedPolygon();
+
     return (
-      <Popover className="geometry-menu-button" key="geometry-menu-button"
-                hasBackdrop={true} position={Position.BOTTOM_RIGHT} >
-        <Button key="geometry-menu-button" icon="menu" minimal={true} />
-        <Menu className="geometry-menu">
-          <MenuItem icon={angleIcon} text={vertexAngleText}
-                    disabled={!enableLabelAngle}
-                    onClick={this.handleToggleVertexAngle}/>
-        </Menu>
-      </Popover>
+      <GeometryToolbarView
+        key="geometry-toolbar-view"
+        model={this.props.model}
+        onAngleLabelClick={this.handleToggleVertexAngle}
+        angleLabelDisabled={!enableVertexAngle}
+        angleLabelSelected={hasVertexAngle}
```

Comment:
One other convention potentially worth following here (in additions to `on/Handle` for callback properties) is that of prefacing boolean properties with `is`, e.g. `isAngleLabelDisabled` and `isAngleLabelSelected`.